packages/transport/src/core/Reference.ts

```diff
@@ -29,14 +31,18 @@ export class Reference extends Serializable implements IReference {
     @serialize()
     public forIdentityTruncated?: string;
 
-    @validate({ nullable: true, min: 1, max: 12, customValidator: (v) => (!Number.isInteger(v) ? "must be an integer" : undefined) })
+    @validate({ nullable: true, regExp: /^(pw|pin(4|5|6|7|8|9|10|11|12|13|14|15|16))$/ })
     @serialize()
-    public passwordType?: number;
+    public passwordType?: string;
+
+    @validate({ nullable: true, customValidator: (v: ICoreBuffer) => (v.buffer.byteLength === 16 ? undefined : "must be 16 bytes long") })
+    @serialize()
+    public salt?: CoreBuffer;
```

Comment:
maybe salt and password type should be one parameter, this would help us to not have to have the validation in the from method.